### PR TITLE
brownfield: Adding more brownfield tests

### PR DESCRIPTION
**Pull request by delqn**
This PR adds more tests to `pkg/brownfield`

Redirects definitely needs more work. Would like to merge the fixtures PR (https://github.com/Azure/application-gateway-kubernetes-ingress/pull/476) to avoid merge conflicts (or semantic merge conflicts)

### CODE REVIEW

File: pkg/brownfield/http_settings_test.go [hunk 1]
```diff
@@ -6,13 +6,28 @@
...
+
+	n "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 
 	ptv1 "github.com/Azure/application-gateway-kubernetes-ingress/pkg/apis/azureingressprohibitedtarget/v1"
 	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/tests/fixtures"
 )
 
+type mockLogger struct {
```
Comment (akshaysngupta) [inline pkg/brownfield/http_settings_test.go]: is this supposed to be in a different file ?

Comment (delqn) [reply pkg/brownfield/http_settings_test.go]: It would actually make sense to move it somewhere in the pkg/tests namespace

Comment (delqn) [reply pkg/brownfield/http_settings_test.go]: Moved to `pkg/tests/mocks`

**Comment by akshaysngupta** [comment]
I think redirect tests can be improved.

**Pull request merged**